pystac/cache.py

```diff
@@ -1,10 +1,15 @@
 from collections import ChainMap
 from copy import copy
+from typing import Any, Dict, List, Optional, TYPE_CHECKING, Tuple, Union, cast
 
-import pystac
+import pystac as ps
 
+if TYPE_CHECKING:
```

Comment:
For type checkers. Those types can't be imported directly because they would cause circular imports. The technique to allow for type signatures to contain types without having to import them at runtime (and so causing a circular import error) is to use the TYPE_CHECKING flag. I've aliased the types with _Type so that it's explicit what is being imported just for type signatures and what is not.